File: output/gemma4-e4b/avatar/Java/Python/codeforces_61_B.py
import re
import sys

def clean_word(s):
    """Removes non-alphabetic characters and converts to lowercase."""
    return re.sub(r'[^a-zA-Z]', '', s).lower()

def solve():
    # Read the three words
    try:
        word1_raw = sys.stdin.readline().strip()
        word2_raw = sys.stdin.readline().strip()
        word3_raw = sys.stdin.readline().strip()
    except EOFError:
        return

    word1 = clean_word(word1_raw)
    word2 = clean_word(word2_raw)
    word3 = clean_word(word3_raw)

    # Generate all 6 permutations
    perms = set()
    perms.add(word1 + word2 + word3)
    perms.add(word1 + word3 + word2)
    perms.add(word2 + word1 + word3)
    perms.add(word2 + word3 + word1)
    perms.add(word3 + word2 + word1)
    perms.add(word3 + word1 + word2)

    # Read the number of students (test cases)
    try:
        students_line = sys.stdin.readline().strip()
        if not students_line:
            return
        students = int(students_line)
    except ValueError:
        return

    # Process test cases
    for _ in range(students):
        try:
            test_case_raw = sys.stdin.readline().strip()
            if not test_case_raw:
                break
            
            test_case = clean_word(test_case_raw)
            
            if test_case in perms:
                print("ACC")
            else:
                print("WA")
        except EOFError:
            break

solve()

File: output/gemma4-e4b/avatar/Python/Java/codeforces_61_B.java
import java.util.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class codeforces_61_B {

    /**
     * Cleans a string by removing characters '-', ';', '_' and converting the rest to lowercase.
     * Equivalent to Python's f(s).
     */
    private static String f(String s) {
        // Remove specified characters and convert to lowercase
        return s.replaceAll("[-;_]", "").toLowerCase();
    }

    /**
     * Generates all permutations of the three input strings and stores the concatenated results.
     * Equivalent to the permutation logic in Python's solve().
     */
    private static Set<String> generatePermutations(String[] l) {
        Set<String> permutations = new HashSet<>();
        
        // Since the length is fixed at 3, we can manually generate all 3! = 6 permutations.
        String s1 = l[0];
        String s2 = l[1];
        String s3 = l[2];

        permutations.add(s1 + s2 + s3);
        permutations.add(s1 + s3 + s2);
        permutations.add(s2 + s1 + s3);
        permutations.add(s2 + s3 + s1);
        permutations.add(s3 + s1 + s2);
        permutations.add(s3 + s2 + s1);
        
        return permutations;
    }

    public static void solve() throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        // 1. Read the three initial strings (Python's input())
        String s1_raw = br.readLine();
        String s2_raw = br.readLine();
        String s3_raw = br.readLine();

        // 2. Apply f() to get the processed strings
        String[] l = new String[3];
        l[0] = f(s1_raw);
        l[1] = f(s2_raw);
        l[2] = f(s3_raw);

        // 3. Generate all permutations and store them in a Set for O(1) lookup
        Set<String> permutations = generatePermutations(l);

        // 4. Read the number of test cases N (Python's inp())
        String nLine = br.readLine();
        if (nLine == null || nLine.isEmpty()) return;
        int N = Integer.parseInt(nLine.trim());

        // 5. Process N test cases
        for (int i = 0; i < N; i++) {
            // Read the input string (Python's instr())
            String inputRaw = br.readLine();
            if (inputRaw == null) break;
            
            // Apply f() to the input
            String processedInput = f(inputRaw);

            // Check if the processed input is in the set of permutations
            if (permutations.contains(processedInput)) {
                System.out.println("ACC");
            } else {
                System.out.println("WA");
            }
        }
    }

    public static void main(String[] args) {
        try {
            solve();
        } catch (IOException e) {
            // Handle potential IO exceptions
        }
    }
}